src/main/java/io/github/elytra/concrete/DefaultMarshallers.java

```diff
@@ -229,18 +237,36 @@ public Number unmarshal(ByteBuf in) {
 		public void marshal(ByteBuf out, Number t) {
 			ByteBufUtils.writeVarInt(out, t.intValue(), 5);
 		}
+	}
+
+	private static class ByteBufMarshaller implements Marshaller<ByteBuf> {
+		@Override
+		public ByteBuf unmarshal(ByteBuf in) {
+			int length = ByteBufUtils.readVarInt(in, 5);
+
+			return in.readBytes(length);
+		}
 
+		@Override
+		public void marshal(ByteBuf out, ByteBuf t) {
+			if (t != null) {
+				ByteBufUtils.writeVarInt(out, t.readableBytes(), 5);
+				out.writeBytes(t.readBytes(t.readableBytes()));
```

Comment:
I tried a couple different things, this worked the best. Writing the internal array doesn't always work, and brings a bunch of junk data along with it. If you can think of something I missed then let me know.